webrtc/browser-to-browser.md

```diff
@@ -0,0 +1,122 @@
+# WebRTC browser-to-browser
+
+| Lifecycle Stage | Maturity      | Status | Latest Revision |
+|-----------------|---------------|--------|-----------------|
+| 1A              | Working Draft | Active | r0, 2022-12-15  |
+
+## Motivation
+
+**Hole punching in the browser** - Enable two browsers or a browser and a non-browser node to connect even though one or both are behind a NAT / firewall.
+
+On a historical note, this specification replaces the existing [libp2p WebRTC star](https://github.com/libp2p/js-libp2p-webrtc-star) and [libp2p WebRTC direct](https://github.com/libp2p/js-libp2p-webrtc-direct) protocols.
+
+## Connection Establishment
+
+### Browser to Browser
+
+Scenario: Browser _A_ wants to connect to Browser node _B_ with the help of server node _R_.
+Both _A_ and _B_ can not listen for incoming connections due to the restriction of the browser platform and being behind a NAT and/or firewall.
+Note that _A_ or _B_ may as well be a non-browser node behind a NAT and/or firewall trying to connect to a browser node.
```

Comment:
```suggestion
Note that _A_ and/or _B_ may as well be a non-browser node behind a NAT and/or firewall trying to connect to a browser node.
```

Reply:
If both A AND B are non-browser nodes I think our current preference would still be to use QUIC or TCP, right? Technically, yes, they could both be non-browser nodes. I don't have an opinion here. Let me know if you still want your suggestion applied @MarcoPolo.

Reply:
Maybe that's our preference, but it's still more correct to say `and/or`, right? There may also be cases where you can't use QUIC, but could use webrtc.

Reply:
:+1: addressed in e1e6a0c. Resolving here. Please comment in case you want this rephrased differently.